crates/fluvio-service/src/server.rs

```diff
@@ -62,159 +39,111 @@ pub trait FlvService {
     ) -> Result<(), SocketError>;
 }
 
-pub struct InnerFlvApiServer<R, A, C, S, T> {
+/// Transform Service into Futures 01
+pub struct FluvioApiServer<R, A, C, S> {
     req: PhantomData<R>,
     api: PhantomData<A>,
     context: C,
     service: Arc<S>,
     addr: String,
-    builder: T,
 }
 
-impl<R, A, C, S, T> fmt::Debug for InnerFlvApiServer<R, A, C, S, T> {
+impl<R, A, C, S> fmt::Debug for FluvioApiServer<R, A, C, S> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "ApiServer({})", self.addr)
+        f.debug_tuple("FluvioApiServer").field(&self.addr).finish()
     }
 }
 
-impl<R, A, C, S, T> InnerFlvApiServer<R, A, C, S, T>
+impl<R, A, C, S> FluvioApiServer<R, A, C, S>
 where
     C: Clone,
 {
-    pub fn inner_new(addr: String, context: C, service: S, builder: T) -> Self {
-        InnerFlvApiServer {
+    pub fn new(addr: String, context: C, service: S) -> Self {
+        FluvioApiServer {
             req: PhantomData,
             api: PhantomData,
             service: Arc::new(service),
             context,
             addr,
-            builder,
         }
     }
 }
 
-pub type FlvApiServer<R, A, C, S> = InnerFlvApiServer<R, A, C, S, DefaultSocketBuilder>;
-
-impl<R, A, C, S> FlvApiServer<R, A, C, S>
-where
-    C: Clone,
-{
-    pub fn new(addr: String, context: C, service: S) -> Self {
-        Self::inner_new(addr, context, service, DefaultSocketBuilder {})
-    }
-}
-
-impl<R, A, C, S, T> InnerFlvApiServer<R, A, C, S, T>
+impl<R, A, C, S> FluvioApiServer<R, A, C, S>
 where
     R: ApiMessage<ApiKey = A> + Send + Debug + 'static,
     C: Clone + Sync + Send + Debug + 'static,
     A: Send + FluvioDecoder + Debug + 'static,
-    S: FlvService<Request = R, Context = C> + Send + Sync + Debug + 'static,
-    T: SocketBuilder + Send + Debug + 'static,
+    S: FluvioService<Request = R, Context = C> + Send + Sync + Debug + 'static,
 {
     pub fn run(self) -> Arc<SimpleEvent> {
-        let event = SimpleEvent::shared();
-        spawn(self.run_shutdown(event.clone()));
-
-        event
+        let shutdown = SimpleEvent::shared();
+        spawn(self.accept_incoming(shutdown.clone()));
+        shutdown
     }
 
-    async fn run_shutdown(self, shutdown_signal: Arc<SimpleEvent>) {
-        match TcpListener::bind(&self.addr).await {
-            Ok(listener) => {
-                debug!(addr = %self.addr, "starting event loop");
-                self.event_loop(listener, shutdown_signal).await;
-            }
+    #[instrument(
+        skip(self, shutdown),
+        fields(addr = %self.addr),
```

Comment:
can remove self and remove add from fields since self implements debug

Reply:
I have come to believe that it's best to put only the very minimal information into the tracing spans because it makes logs very noisy. I think that especially since this is at a top-level hierarchy that a lot of other code paths stem from, it'd be good to keep noise down at this point

Reply:
> it is more important to keep consistent so have reliable trace which I found to be critical.  Noise problem can be delegated to tooling/filter